src/database.rs

```diff
@@ -513,6 +516,31 @@ impl Database {
         self.verify.as_ref().unwrap()
     }
 
+    /// Get a statement by label.
+    pub fn statement(&mut self, name: &str) -> Option<StatementRef> {
+        match self.name_result().lookup_label(name.as_bytes()) {
+            None => None,
+            Some(lookup) => Some(self.parse_result().statement(lookup.address)),
+        }
+    }
+
+    /// Export an mmp file for a given statement.
+    pub fn export(&mut self, stmt: String) {
+        time(&self.options.clone(), "export", || {
+            let parse = self.parse_result().clone();
```

Comment:
This is pretty meh.  I might switch to using interior mutability, so that `foo_result` can take self by shareable reference and then you won't need quite so many temporary variables.  I think the original reason I didn't was a desire to keep `Database` `Send+Sync`, but that's a little silly if everything needs unique refs; a version with interior cells would still be `Send+Clone`, which is about as useful to us